dallinger/config.py

```diff
@@ -102,6 +103,10 @@ def load_from_environment(self):
     def load_config(self):
         if self.ready:
             raise ValueError("Already loaded")
+
+        # Apply extra settings before loading the configs
+        self.apply_extra_settings()
```

Comment:
I think "register" may be a less confusing verb than "apply" here. Maybe `register_experiment_settings` instead of `apply_extra_settings`?